src/NServiceBus.Core/Transports/TransportReceiver.cs

```diff
@@ -61,14 +62,25 @@ class TransportReceiver
 
         async Task InvokePipeline(PushContext pushContext)
         {
+            //note: we should get this value from the transport in the future to make it more accurate
+            var pipelineStartedAt = DateTime.UtcNow;
+
+
             using (var childBuilder = builder.CreateChildBuilder())
             {
+                var sw = new Stopwatch();
+
+                sw.Start();
+
                 var rootContext = new RootContext(childBuilder, pipelineCache, eventAggregator);
-                var context = new TransportReceiveContext(new IncomingMessage(pushContext.MessageId, pushContext.Headers, pushContext.BodyStream), pushContext.TransportTransaction, pushContext.ReceiveCancellationTokenSource, rootContext);
+                var message = new IncomingMessage(pushContext.MessageId, pushContext.Headers, pushContext.BodyStream);
+                var context = new TransportReceiveContext(message, pushContext.TransportTransaction, pushContext.ReceiveCancellationTokenSource, rootContext);
                 context.Extensions.Merge(pushContext.Context);
+
+
                 await pipeline.Invoke(context).ConfigureAwait(false);
 
-                await context.RaiseNotification(new ReceivePipelineCompleted()).ConfigureAwait(false);
+                await context.RaiseNotification(new ReceivePipelineCompleted(message, pipelineStartedAt,sw.Elapsed)).ConfigureAwait(false);
```

Comment:
Correct me if I'm wrong, but the continuation may be run on a different thread. If yes, then, accessing `Elapsed` may return not precise result (see [this StackOverflow question](http://stackoverflow.com/a/17505960)).